app/src/ui/components/offers/CreateOffer.vue

```diff
@@ -65,17 +72,24 @@ document.documentElement.style.setProperty('--vh', `${vh}px`)
 function calculateMarginRate() {
   rate.value = convertMarginRateToOfferRate(margin.value, marginOffsetUnmasked.value)
 }
-function createOffer() {
+async function createOffer() {
+  const telegramHandle = removeTelegramURLPrefix(ownerContact.value)
+  console.log('owner_contact: ', telegramHandle)
+  // Encrypt contact to save on the profile when an offer is created
+  const owner_contact_pk = secrets.value.publicKey
+  const owner_contact = await encryptData(owner_contact_pk, telegramHandle)
+
   const postOffer: PostOffer = {
-    owner_contact: removeTelegramURLPrefix(ownerContact.value),
+    owner_contact,
+    owner_encrypt_key: owner_contact_pk,
```

Comment:
I think that `owner_encryption_key` or `owner_contact_pubkey` would be better names for these values, also I would rename one of these two to match the other so they're the same name.

Reply:
Yeah ... still in development, I will change to better names